auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/AdminEnableDisableUserController.kt

```diff
@@ -0,0 +1,101 @@
+package uk.gov.communities.delta.auth.controllers.internal
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.util.*
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.AzureADSSOConfig
+import uk.gov.communities.delta.auth.config.DeltaConfig
+import uk.gov.communities.delta.auth.config.LDAPConfig.Companion.VALID_USERNAME_REGEX
+import uk.gov.communities.delta.auth.plugins.ApiError
+import uk.gov.communities.delta.auth.repositories.LdapUser
+import uk.gov.communities.delta.auth.services.SetPasswordTokenService
+import uk.gov.communities.delta.auth.services.UserAuditService
+import uk.gov.communities.delta.auth.services.UserLookupService
+import uk.gov.communities.delta.auth.services.UserService
+import uk.gov.communities.delta.auth.utils.randomBase64
+import javax.naming.NameNotFoundException
+
+class AdminEnableDisableUserController(
+    private val ssoConfig: AzureADSSOConfig,
+    private val userLookupService: UserLookupService,
+    private val userService: UserService,
+    private val setPasswordTokenService: SetPasswordTokenService,
+    private val auditService: UserAuditService,
+) : AdminUserController(userLookupService) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    private val allowedRoles =
+        arrayOf(DeltaConfig.DATAMART_DELTA_READ_ONLY_ADMIN, DeltaConfig.DATAMART_DELTA_ADMIN)
+
+    suspend fun enableUser(call: ApplicationCall) {
+        val session = getSessionIfUserHasPermittedRole(allowedRoles, call)
+
+        val user = findUserFromCallParameter(call)
+
+        if (user.accountEnabled) {
+            return call.respond(mapOf("message" to "Account already enabled"))
+        }
+
+        if (user.passwordLastSet == null) {
+            if (user.isSSORequiredUser()) {
+                logger.atInfo().addKeyValue("UserCN", user.cn)
```

Comment:
Really minor, and maybe not an issue, but something like UserCN might be interpreted as the requesting user rather than the target user. 

AdminEmailController adds a key of "userCNToSendEmailTo" to some logs. I think maybe targetUserCN or similar might be fine, but 🤷‍♂️

Reply:
Makes sense, I'll change it